pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -820,26 +819,73 @@ func (r *Builder) TemplateLabels(vmRef ref.Ref) (labels map[string]string, err e
 	default:
 		os = UnknownOS
 	}
+	return
+}
 
+func (r *Builder) getPreferenceOs(vm *model.Workload, os, version, distro string) string {
```

Comment:
how about adding unit tests for this?

Reply:
i'll try to do on another commit, i changed all the rest first